packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/tera-calibrate-ciemss-drilldown.vue

```diff
@@ -735,285 +715,58 @@ const lossChartSize = useDrilldownChartSize(lossChartContainer);
-			label: variable
-		}));
-	return variables;
-});
-
-const errorChart = computed(() => {
-	if (errorData.value.length === 0) return {};
-	const spec = createErrorChart(errorData.value, {
-		title: '',
-		width: chartSize.value.width,
-		variables: errorChartVariables.value,
-
-		xAxisTitle: 'Mean absolute (MAE)'
-	});
-	return spec;
-});
-
-const onExpandErrorChart = () => {
-	if (errorData.value.length === 0) return {};
-	// Customize the chart size by modifying the spec before expanding the chart
-	const spec = createErrorChart(errorData.value, {
-		title: '',
-		width: window.innerWidth / 1.5,
-		height: 230,
-		boxPlotHeight: 50,
-		areaChartHeight: 150,
-		variables: errorChartVariables.value,
-		xAxisTitle: 'Mean absolute (MAE)'
-	});
-	return spec as any;
-};
-
-const comparisonCharts = computed(() => {
-	const charts: Record<string, any> = {};
-	if (!preparedChartInputs.value) return charts;
-	const { result, resultSummary, reverseMap } = preparedChartInputs.value;
-	selectedComparisonChartSettings.value.forEach((setting) => {
-		const selectedVars = setting.selectedVariables;
-		const { statLayerVariables, sampleLayerVariables, options } = createForecastChartOptions(setting, reverseMap);
-		const annotations = getChartAnnotationsByChartId(setting.id);
-
-		const chart = applyForecastChartAnnotations(
-			createForecastChart(
-				{
-					data: result,
-					variables: sampleLayerVariables,
-					timeField: 'timepoint_id',
-					groupField: 'sample_id'
-				},
-				{
-					data: resultSummary,
-					variables: statLayerVariables,
-					timeField: 'timepoint_id'
-				},
-				null,
-				options
-			),
-			annotations
-		);
-		if (interventionPolicy.value) {
-			_.keys(groupedInterventionOutputs.value).forEach((key) => {
-				if (selectedVars.includes(key)) {
-					chart.layer.push(...createInterventionChartMarkers(groupedInterventionOutputs.value[key]));
-				}
-			});
-		}
-		charts[setting.id] = chart;
-	});
-	return charts;
-});
+const preparedChartInputs = usePreparedChartInputs(
+	props,
+	runResult,
+	runResultSummary,
+	runResultPre,
+	runResultSummaryPre
+);
+const pyciemssMap = computed(() => preparedChartInputs.value?.pyciemssMap ?? {});
+const {
+	activeChartSettings,
+	chartSettings,
+	selectedVariableSettings,
+	selectedParameterSettings,
+	selectedInterventionSettings,
+	selectedErrorVariableSettings,
+	selectedComparisonChartSettings,
+	comparisonChartsSettingsSelection,
+	removeChartSettings,
+	updateChartSettings,
+	addComparisonChartSettings
+} = useChartSettings(props, emit);
+
+const {
+	generateAnnotation,
+	getChartAnnotationsByChartId,
+	useInterventionCharts,
+	useVariableCharts,
+	useComparisonCharts,
+	useErrorChart,
+	useParameterDistributionCharts
+} = useCharts(
+	props.node.id,
+	model,
+	modelConfig,
+	preparedChartInputs,
+	chartSize,
+	computed(() => interventionPolicy.value?.interventions ?? [])
+);
+const toDatasetVar = (modelVar: string) => mapModelVarToDatasetVar(props.node.state, modelVar);
```

Comment:
I am not 100% sure if this is exclusively used for output?

If this is then I suspect we want to use the mapping from our output similar to my other comment

Reply:
Done. I've change the codes so that we grab the mapping from output settings